desktop/apps/gene/client.js

```diff
@@ -19,17 +23,39 @@ const relatedArtistsTemplate = (args) => {
   return require('./templates/related_artists.jade')(args)
 }
 
+// Update URL with current filters/mode/sort, for ease of sharing.
+const onStateChange = ({ filters, sort, mode }) => {
+  let params
+  if (mode === 'artists') {
+    params = assign({ mode }, {})
+  } else {
+    params = assign({ ...filters, sort, mode }, {})
+  }
+  const fragment = '?' + qs.stringify(params)
+  window.history.replaceState({}, new Gene(sd.GENE).toPageTitle(), fragment)
+}
+
 function setupGenePage () {
-  // Init GenePage component
+  // Pull out sort and filters from URL, if present
   const urlParams = qs.parse(location.search.replace(/^\?/, ''))
-  const options = Object.assign({}, urlParams, { geneID: sd.GENE.id })
+  let sort, mode
+  if (urlParams.sort) {
+    sort = urlParams.sort
+    delete urlParams.sort
+  }
+  if (urlParams.mode) {
+    mode = urlParams.mode
+    delete urlParams.mode
+  } else {
+    mode = sd.MODE
+  }
 
-  init({
-    user: sd.CURRENT_USER,
-    component: components.Gene,
-    domID: 'gene-filter',
-    queryConfig: new configs.GeneQueryConfig(options)
-  })
+  const options = Object.assign({}, { sort }, { filters: { ...urlParams } }, { geneID: sd.GENE.id, mode })
```

Comment:
Why do you use `Object.assign(…)` here? Can’t you just create 1 object immediately instead of merging?

Reply:
I'm using `Object.assign` to create the new object....I've seen this pattern everywhere.

Reply:
Merging obviously definitely has its place, but here you are creating 4 new objects and then merging them. As far as I can tell this is the equivalent of:

```js
 { sort, filters: { ...urlParams }, geneID: sd.GENE.id, mode }
```

Am I overlooking/misunderstanding something?

Reply:
No. My point was just that I've noticed we use this pattern (`assign`), all over, even in places where I would agree with you, it just seems like we're creating a new object.

So I just decided to follow the existing pattern. It doesn't matter to me, I can switch.

Reply:
I think the code is simpler to read without it. And while a single instance won’t be too bad, if this is really happening all over the place then at some point the overhead of all these objects will add up.